scripts/make_COMPASS_ELM_forcing.py

```diff
@@ -1,247 +1,154 @@
+    site_id = row.site_id
+
+    # Subset forcing to current site
+    hydro_forcing_subset = hydro_filled_df[hydro_filled_df.site_name == row.site_name]
+
+    # Write to xarray
+    tide_data_multicell['tide_height'][:, index]  = np.array(hydro_forcing_subset['water_height_m'])
+    tide_data_multicell['tide_salinity'][:, index]= np.array(hydro_forcing_subset['water_salinity'])
+
+
+#--------------------------------------------------------------------------------------
+#  MULTI-CELL SURFACE
+# Surface data set below will define ground surface height above drainage to give different hydrological conditions.
+# Make new multi-grid cell configuration. Treating each land cover type as a separate grid cell
+# This is easier for setting up tidal forcing, but won't work in a larger scale simulation where grid cells need to be spatially defined
+# Long term solution is to use topo units, but will need to figure a way to do hydro forcing in that framework
+# Here we start from the single grid cell configuration for the site and then multiply it into multiple grid cells
+
+domain_threecol  =xr.open_dataset('../data/raw/surface_data/domain_3col.nc')
+surfdata_threecol=xr.open_dataset('../data/raw/surface_data/surfdata_3col.nc')
+surfdata_onecol  =xr.open_dataset('../data/raw/surface_data/surfdata_1x1pt_US-GC_TransTEMPEST_c20230901.nc')
+# landuse_onecol=xarray.open_dataset(f'/nfs/data/ccsi/proj-shared/E3SM/pt-e3sm-inputdata/atm/datm7/GSWP3_daymet/cpl_bypass_{site}/surfdata.pftdyn.nc')
+# TODO: NLCD FOR LAND USE
+
+# Threecol is 3 points along transect, multiplying by number of sites for sites x 3 total points
+domain_multicell=xr.concat([domain_threecol] * len(synoptic.site_id.unique()),dim='ni')
+# Assume all grid cells are the same size.
+cell_width=0.02
+
+# Loop through grid-points
+for n, row in synoptic.iterrows():
+    lat = row.lat
+    lon = row.long
+    # Write lat to file
+    domain_multicell['yc'][0,n]=lat
+    # Fix long coordinates to a 0-360deg format
+    if lon>0:
+        domain_multicell['xc'][0,n]=lon
+    else:
+        domain_multicell['xc'][0,n]=lon+360
+    # Write in elevation and distance
+    surfdata_multicell['ht_above_stream'].squeeze()[n] = row.elev
+    surfdata_multicell['dist_from_stream'].squeeze()[n] = row.distance
+
+# Modify coordinates
+domain_multicell['xv'][0,:,[0,2]] =  domain_multicell['xc'].T.squeeze() + cell_width/2
+domain_multicell['xv'][0,:,[1,3]] =  domain_multicell['xc'].T.squeeze() - cell_width/2
+domain_multicell['yv'][0,:,[0,2]] =  domain_multicell['yc'].T.squeeze() + cell_width/2
+domain_multicell['yv'][0,:,[1,3]] =  domain_multicell['yc'].T.squeeze() - cell_width/2
+
+surfdata_multicell = xr.concat([surfdata_onecol]*len(synoptic.site_id.unique())*3,dim='lsmlon',data_vars='minimal')
+surfdata_multicell['LONGXY'][:]=domain_multicell['xc'].values
+surfdata_multicell['LATIXY'][:]=domain_multicell['yc'].values
+surfdata_multicell['ht_above_stream'] = surfdata_multicell['TOPO']
```

Comment:
This line and the next is overwriting where you set the land surface height and distance above

Reply:
Good catch.  I modified the order of operations so that `ht_above_stream` and `dist_from_stream` are created/renamed from `TOPO`, and only then are replaced by the correct values, site-by-site.